dance/transforms/filter.py

```diff
@@ -884,3 +885,83 @@ def gini_func(x, weights=None):
+                       ['max_cells', 'min_counts', 'min_cells', 'max_counts'],
+                       ['max_cells', 'min_counts', 'max_counts', 'min_cells'],
+                       ['max_cells', 'min_cells', 'min_counts', 'max_counts'],
+                       ['max_cells', 'min_cells', 'max_counts', 'min_counts'],
+                       ['max_cells', 'max_counts', 'min_counts', 'min_cells'],
+                       ['max_cells', 'max_counts', 'min_cells', 'min_counts']]
+
+
+@register_preprocessor("filter", "gene")
+class FilterGenesScanpyOrder(BaseTransform):
+    """Scanpy filtering gene transformation with additional options.
+
+    Parameters
+    ----------
+    order_index
+        Index of (min_counts,min_cells,max_counts,max_cells) order
+    min_counts
+        Minimum number of counts required for a gene to be kept.
+    min_cells
+        Minimum number (or ratio) of cells required for a gene to be kept.
+    max_counts
+        Maximum number of counts required for a gene to be kept.
+    max_cells
+        Maximum number (or ratio) of cells required for a gene to be kept.
+    split_name
+        Which split to be used for filtering.
+    channel
+        Channel to be used for filtering.
+    channel_type
+        Channel type to be used for filtering.
+
+    """
+
+    def __init__(self, order_index: int, min_counts: Optional[int] = None,
+                 min_cells: Optional[Union[float, int]] = None, max_counts: Optional[int] = None,
+                 max_cells: Optional[Union[float, int]] = None, split_name: Optional[str] = None,
+                 channel: Optional[str] = None, channel_type: Optional[str] = "X", **kwargs):
+        if order_index < 0 or order_index >= len(filter_genes_orders):
+            raise KeyError(f"An integer between 0 and {len(filter_genes_orders)-1} should be chosen")
+        self.filter_genes_order = filter_genes_orders[order_index]
+        self.logger.info(f"choose filter_genes_order f{self.filter_genes_order}")
+        self.geneScanpyOrderDict = {
+            "min_counts":
+            FilterGenesScanpy(min_counts=min_counts, split_name=split_name, channel=channel, channel_type=channel_type,
+                              **kwargs),
+            "min_cells":
+            FilterGenesScanpy(min_cells=min_cells, split_name=split_name, channel=channel, channel_type=channel_type,
+                              **kwargs),
+            "max_counts":
+            FilterGenesScanpy(max_counts=max_counts, split_name=split_name, channel=channel, channel_type=channel_type,
+                              **kwargs),
+            "max_cells":
+            FilterGenesScanpy(max_cells=max_cells, split_name=split_name, channel=channel, channel_type=channel_type,
+                              **kwargs)
+        }
+
+    def __call__(self, data: Data):
+        for parameter in self.filter_genes_order:
+            geneScanpyOrder = self.geneScanpyOrderDict[parameter]
+            geneScanpyOrder(data)
```

Comment:
Is there way to only use one filtering instead of all four?

Reply:
It is possible to add missing cases to orders in yml, and now the test has been successful, but then there will be invalid missing numbers, so the best practice is to use all four numbers, set min_counts and min_cells to 0, and max_counts and max_cells to a maximum